### PR TITLE
[backport] Fix tests for incompatible behavior of NumPy 1.20 `linspace`

### PR DESCRIPTION
**Pull request by asi1024**
Backport of https://github.com/cupy/cupy/pull/4604

### CODE REVIEW

File: tests/cupy_tests/creation_tests/test_ranges.py [hunk 1]
```diff
@@ -157,7 +157,7 @@ def test_linspace_array_start_stop(self, xp, dtype_range, dtype_out):
         stop = xp.array([100, 0], dtype=dtype_range)
         return xp.linspace(start, stop, num=50, dtype=dtype_out)
 
-    @testing.with_requires('numpy>=1.16')
+    @testing.with_requires('numpy>=1.16<1.20')
```
Comment (toslunar) [inline tests/cupy_tests/creation_tests/test_ranges.py:160]:
```suggestion
    @testing.with_requires('numpy>=1.16,<1.20')
```

File: tests/cupy_tests/creation_tests/test_ranges.py [hunk 2]
```diff
@@ -169,7 +169,7 @@ def test_linspace_mixed_start_stop(self, xp, dtype_range, dtype_out):
             stop = xp.array([100, -100], dtype=dtype_range)
         return xp.linspace(start, stop, num=50, dtype=dtype_out)
 
-    @testing.with_requires('numpy>=1.16')
+    @testing.with_requires('numpy>=1.16<1.20')
```
Comment (toslunar) [inline tests/cupy_tests/creation_tests/test_ranges.py:172]:
```suggestion
    @testing.with_requires('numpy>=1.16,<1.20')
```

**Review by toslunar — CHANGES_REQUESTED** [review]
```
>>> req, = pkg_resources.parse_requirements('numpy>=1.16<1.20'); req.specs
[('>=', '1.16<1.20')]
>>> req, = pkg_resources.parse_requirements('numpy>=1.16,<1.20'); req.specs
[('<', '1.20'), ('>=', '1.16')]
```

**Comment by asi1024** [comment]
Jenkins, test this please.

**Review by toslunar — APPROVED** [review]

**Pull request merged**